firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/RemoteSettings.kt

```diff
@@ -47,45 +61,100 @@ internal class RemoteSettings(val context: Context) : SettingsProvider {
     }
 
   override fun updateSettings() {
-    fetchConfigs()
+    runBlocking { launch(Dispatchers.Default) { fetchConfigs() } }
   }
 
   override fun isSettingsStale(): Boolean {
     return settingsCache.hasCacheExpired()
   }
 
-  companion object SettingsFetcher {
-    private const val SESSION_CONFIGS_NAME = "firebase_session_settings"
-    private const val FIREBASE_SESSIONS_BASE_URL_STRING =
-      "https://firebase-settings.crashlytics.com"
-    private const val FIREBASE_PLATFORM = "android"
-    private const val fetchInProgress = false
-    private val settingsUrl: URL = run {
-      var uri =
-        Uri.Builder()
-          .scheme("https")
-          .authority(FIREBASE_SESSIONS_BASE_URL_STRING)
-          .appendPath("spi/v2/platforms")
-          .appendPath(FIREBASE_PLATFORM)
-          .appendPath("gmp")
-          // TODO(visum) Replace below with the GMP APPId
-          .appendPath("GMP_APP_ID")
-          .appendPath("settings")
-          .appendQueryParameter("build_version", "")
-          .appendQueryParameter("display_version", "")
-
-      URL(uri.build().toString())
+  internal fun clearCachedSettings() {
+    runBlocking { launch(Dispatchers.Default) { settingsCache.removeConfigs() } }
+  }
+
+  suspend private fun fetchConfigs() {
+    // Check if a fetch is in progress. If yes, return
+    if (fetchInProgress) {
```

Comment:
Does fetchInProgress need to be synchronized?

Reply:
Thanks for flagging it. Made it an `AtomicBoolean` to protect it.